iceoryx_posh/test/moduletests/test_roudi_portpool.cpp

```diff
@@ -0,0 +1,604 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/internal/roudi/port_pool_data.hpp"
+#include "iceoryx_posh/internal/runtime/node_data.hpp"
+#include "iceoryx_posh/popo/subscriber_options.hpp"
+#include "iceoryx_posh/roudi/port_pool.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using ::testing::Return;
+using namespace iox::capro;
+
+namespace iox
+{
+namespace test
+{
+static constexpr uint32_t DEFAULT_DEVICE_ID{0u};
```

Comment:
Please initialize here with values other than null to ensure that the passed parameters are correctly taken.

Reply:
done